sycl/include/sycl/ext/oneapi/matrix/static-query.hpp

```diff
@@ -463,21 +368,21 @@ struct tpu_params<tpu::dpas, Ta, Tb, Tc, 0, 0, 0,
     uint32_t nsize;
     uint32_t ksize;
   };
-  static constexpr int num_combinations = 4;
-  combination combinations[num_combinations];
-  constexpr tpu_params() : combinations() {
-    int i = 0;
-    combinations[i].msize = 1;
-    combinations[i + 1].msize = 2;
-    combinations[i + 2].msize = 4;
-    combinations[i + 3].msize = 8;
-    for (int ii = 0; ii < 4; ii++) {
-      combinations[i + ii].ksize = ((sizeof(Ta) == 1) ? 32 : 16);
-      combinations[i + ii].nsize = 8;
-    }
-  }
+  static constexpr combination combinations[] = {
+      {0, 0, 0, matrix_type(NULL), matrix_type(NULL), matrix_type(NULL), 1, 8,
```

Comment:
This "matrix_type(NULL)" looks a bit odd to me, and it might look odd to compiler as well, causing warnings (but I am not sure). "matrix_type(NULL)" means matrix_type::bf8, right?

Probably, this place is the exception and the older initialization looked better here. @dm-vodopyanov , @bashbaug - what do you think about this?

Reply:
@v-klochkov With this new initialization, "matrix_type(NULL)" is what passed the compilation. And it is just NULL, no value.
The other way to fix this is to check "Ta" against each value of matrix_type enum.

I cannot go back to the old initialization because I removed the constructor. And since I need constexpr, with constexpr, you have to initialize the array right away.

Reply:
> And it is just NULL, no value.

NULL is a value, normally an integral value: https://en.cppreference.com/w/cpp/types/NULL
That NULL is treated as 0 and 0 is converted to matrix_type::bf8.

Maybe just use matrix_type::bf8 there and write a comment saying that the types used in the initialization below are fake and not used?

```
int main() {
  using mt = matrix_type;
  combination combinations[] = {
      {16, 16, 64, mt::sint8, mt::sint8, mt::sint32},
      {16, 16, 64, matrix_type(NULL), mt::uint8, mt::sint32}
  };
  std::cout << "is NULL equal to bf8: " << (combinations[1].atype == matrix_type::bf8) << std::endl;
}

```

will print:
`is NULL equal to bf8: 1`

Reply:
done